Assignment6.java

```diff
@@ -90,16 +88,17 @@ private static int nextCarToMove(int[] given, int[] desired, Map<Integer, Intege
         
         //car belonging to the current empty slot
         if (desired[indexMap.get(0)] != 0) {
-            return desired[indexMap.get(0)];
+            // remove car from set of misplaced cars
+            int nextCarToMove = desired[indexMap.get(0)];
+            misplacedCars.remove(nextCarToMove);
+            return nextCarToMove;
 
         }
 
         //empty lot in its desired place => deblock the algorithm
         // pick a random car from set of misplaced cars
         if (misplacedCars.size() > 0) {
-            Integer nextCarToMove = misplacedCars.iterator().next();
-            misplacedCars.remove(nextCarToMove);
-            return nextCarToMove;
+            return misplacedCars.iterator().next();

```

Comment:
oh I see what you mean. I didn't follow yesterday's discussion, it was too late for me. Makes sense now, thanks!